### PR TITLE
move UI tests to drone

### PR DESCRIPTION
**Pull request by individual-it**
1. some improvements on the UI tests infrastructure (was needed because we run on different timing issues in drone - drone is just sooo fast)
2. run the UI tests in drone

ToDo:

- [x] change from `arturneumann/owncloud-ci-php` to `owncloudci/php` as soon as its build the version after https://github.com/owncloud-ci/php/pull/13

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -120,6 +120,78 @@ pipeline:
       matrix:
         TEST_SUITE: files_external
 
+  install-server:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    commands:
+      - cd /drone/src/
+      - ls -l
+      - php ./occ maintenance:install -vvv --database=sqlite --database-name=owncloud --database-table-prefix=oc_ --admin-user=admin --admin-pass=admin --data-dir=/drone/src/data
```
Comment (tboerger) [inline .drone.yml]: I would say use `./occ ...` or `php occ ...` ;)

File: .drone.yml [hunk 2]
```diff
@@ -120,6 +120,78 @@ pipeline:
...
+      - php ./occ a:l
+      - php ./occ a:e testing
+      - php ./occ a:l
+      - php ./occ config:system:set trusted_domains 1 --value=php-dev-server
+      - ls -l
+    when:
+      matrix:
+        TEST_SUITE: selenium
+
+  owncloud-log:
```
Comment (tboerger) [inline .drone.yml]: Maybe move that simply to services? tail should have a flag to not fail if the file doesn't exist yet.

File: .drone.yml [hunk 3]
```diff
@@ -120,6 +120,78 @@ pipeline:
       matrix:
         TEST_SUITE: files_external
 
+  install-server:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    commands:
+      - cd /drone/src/
+      - ls -l
```
Comment (tboerger) [inline .drone.yml]: Why do you add this listing?

File: .drone.yml [hunk 4]
```diff
@@ -120,6 +120,78 @@ pipeline:
...
+    when:
+      matrix:
+        TEST_SUITE: selenium
+
+  owncloud-log:
+    image: owncloud/ubuntu:16.04
+    detach: true
+    pull: true
+    commands:
+      - whoami
```
Comment (tboerger) [inline .drone.yml]: Maybe simply tail the log and that's it? `whoami` should anyway always be `root`, and the other files within the listing are anyway part of the repo?

File: .drone.yml [hunk 5]
```diff
@@ -120,6 +120,78 @@ pipeline:
...
+      - ls -l /drone/src/
+      - ls -l /drone/src/data/
+      - ls -l /drone/src/data/owncloud.log
+      - tail -f /drone/src/data/owncloud.log
+    when:
+      matrix:
+        TEST_SUITE: selenium
+
+  php-dev-server:
```
Comment (tboerger) [inline .drone.yml]: Just name it `server`? And maybe move it to services?

File: .drone.yml [hunk 6]
```diff
@@ -120,6 +120,78 @@ pipeline:
...
+    detach: true
+    pull: true
+    commands:
+      - cd /drone/src/
+      - php -S 0.0.0.0:80
+    when:
+      matrix:
+        TEST_SUITE: selenium
+
+  remote-fed-php-dev-server:
```
Comment (tboerger) [inline .drone.yml]: As above, maybe simply name it `federated`? And maybe move it to services?

**Review by tboerger — COMMENTED** [review]
Generally looks awesome, my comments are so far only NITPICKing.

File: .drone.yml (hunk 3 above)
Comment (individual-it) [reply .drone.yml]: just for debugging, will delete it

File: .drone.yml (hunk 2 above)
Comment (individual-it) [reply .drone.yml]: I want it to start after `install-server` that's why I added it here and used `detach: true`

**Comment by individual-it** [comment]
Drone is green now, please review.
We will have to see the next time how stable it works.

File: .drone.yml [hunk 7]
```diff
@@ -180,6 +228,34 @@ services:
...
+    when:
+      matrix:
+        TEST_SUITE: selenium
+
+  server:
+    image: owncloudci/php:latest
+    pull: true
+    environment:
+      - APACHE_WEBROOT=/drone/src/
+    entrypoint: [ "/bin/bash" ]
```
Comment (tboerger) [inline .drone.yml]: Do we really need this entrypoint? It should be enough to directly call Apachectl

File: .drone.yml [hunk 8]
```diff
@@ -180,6 +228,34 @@ services:
...
+    when:
+      matrix:
+        TEST_SUITE: selenium
+
+  server:
+    image: owncloudci/php:latest
+    pull: true
+    environment:
+      - APACHE_WEBROOT=/drone/src/
+    entrypoint: [ "/usr/local/bin/apachectl" ]
```
Comment (tboerger) [inline .drone.yml]: I was thinking more about something like:

```
    command:  [ "/usr/local/bin/apachectl", "-e debug" , "-DFOREGROUND" ]
```

I have not checked it but maybe you can even skip the absolute path and simply call `["apachectl", "-e", "debug", "-DFOREGROUND"]`

Comment (individual-it) [reply .drone.yml]: without the absolute path it would call `/usr/sbin/apachectl`

File: .drone.yml [hunk 9]
```diff
@@ -180,6 +228,32 @@ services:
...
+
+  server:
+    image: owncloudci/php:latest
+    pull: true
+    environment:
+      - APACHE_WEBROOT=/drone/src/
+    command: [ "/usr/local/bin/apachectl", "-e debug" , "-DFOREGROUND" ]
+    when:
+      matrix:
+        TEST_SUITE: selenium
```
Comment (patrickjahns) [inline .drone.yml]: we should maybe introduce a general "FLAG" like `USE_SERVER` and `USE_FEDERATED_SERVER` and just define when to use the server and federated in the matrix part? 

I.e.
```
    - PHP_VERSION: 7.1
      TEST_SUITE: selenium
      BEHAT_SUITE: other
      DB_TYPE: mariadb
      USE_SERVER: true
      USE_FEDERATED_SERVER: true
```

just a thought

**Comment by patrickjahns** [comment]
What about having drone run in parallel to travis for a week and see our learnings here?

In the next steps we could also introduce running firefox in drone (if required) and defer running Internet Explorer / Edge on travis. (Even defer further to run on nightly?)

File: .drone.yml (hunk 9 above)
Comment (individual-it) [reply .drone.yml]: done

**Comment by individual-it** [comment]
> What about having drone run in parallel to travis for a week and see our learnings here?

travis does not run UI tests on PRs for master currently, so enabling this would be the only UI tests done for master. It should not do any harm to merge except of if too many builds fail and devs. get annoyed. The problem in drone is that you cannot restart a single task, but only the whole build

**Comment by individual-it** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Review by patrickjahns — APPROVED** [review]

**Comment by phil-davis** [comment]
Backport stable10 #30209

**Pull request merged**